WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php

```diff
@@ -914,6 +922,16 @@ private function validate_prefixes() {
 				continue;
 			}
 
+			if ( mb_strlen( $prefix ) < self::MIN_PREFIX_LENGTH ) {
```

Comment:
PHP_CodeSniffer itself uses `iconv` instead `MBString` when examining potentially multi-byte text. IIRC, this is because the `iconv` extension is more widely installed.
The results of using `iconv` would be the same: https://3v4l.org/bejLK

For the `iconv` equivalent, see: https://www.php.net/manual/en/function.iconv-strlen.php

Whichever is used, you cannot rely on the `iconv` or `MBstring` extension being available, so a function call to either should always be accompanied by a call to `function_exists()` or `extension_loaded()`.

Similarly, whichever is used, you will need to pass the `$encoding`.
Since PHPCS 3.0, the default encoding presumed is `utf-8`, but projects can change this using `--encoding=...`.

To get the correct encoding, use `$phpcsFile->config->encoding`.

> I've implemented mb_strlen() instead of strlen() and added tests. PHP <=5.5 tests now break because mb_strlen() has a different outcome.

Setting the `$encoding` should fix this. The default `encoding` used by PHP for MbString changed in PHP 5.6 to `utf-8` which explains why the unit tests pass on PHP 5.6 and higher and fail below.

Reply:
Unrelated to this PR, but I see a potential issue with the `strtolower()` on line 913 in this context as it doesn't take multibyte characters into account either.

Reply:
FYI: https://github.com/PHPCompatibility/PHPCompatibility/issues/839 :wink:

Reply:
Very interesting. I've converted the check to use the`iconv_strlen` function, use the correct encoding given by the `phpcsFile` and check for availability of the function.

The `strtolower` shouldn't cause to much of an issue, as it's only used to check against the blacklist, and for now there are no non-ascii characters found in the blacklist.